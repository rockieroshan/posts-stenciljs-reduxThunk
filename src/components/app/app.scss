@import '../../assets/scss/main.scss';

.appRoot {
  @include flexCenter;
  max-width: 1200px;
  margin: 0 auto;

  input {
    width: 360px;
  }

  .mainWrapper {
    margin: 0 2em 3em 2em;
    @include flexCenter;
  }

  @include screenType(tablet) {
    @include flexColumn;
    overflow: hidden;

    .mainWrapper {
      @include flexColumn;
    }
  }
}
