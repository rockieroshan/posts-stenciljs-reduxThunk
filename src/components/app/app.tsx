import { Component, h } from '@stencil/core';
import configureStore from '../../store';
import { store } from '@stencil/redux';

@Component({
  tag: 'app-root',
  styleUrl: 'app.scss',
  shadow: true,
})

export class AppRoot {
  componentWillLoad() {
    store.setStore(configureStore({}));
  }

  render() {
    return (
      <div class='appRoot'>
        <div class='mainWrapper'>
          {/* <app-editPost /> */}
          <stencil-router id="router">
            <stencil-route
              url="/edit/:id"
              component="app-editcard"
              exact={true}
            />
            <stencil-route
              url="/"
              component="app-posts"
              exact={true}
            />
          </stencil-router>
        </div>
      </div>
    );
  }
}
