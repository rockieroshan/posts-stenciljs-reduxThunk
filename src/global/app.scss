@import '../assets/scss/main.scss';

html {
    font-size: 16px;

    body {
        margin: 0;
        background: $backgroundColor;
        font-family: 'Gill Sans', sans-serif;
        box-sizing: border-box;
        overflow: auto;
        height: -webkit-fill-available;
    }
}
