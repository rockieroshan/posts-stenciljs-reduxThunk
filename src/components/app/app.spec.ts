
import { newSpecPage } from '@stencil/core/testing';
import { AppRoot } from './app';

describe('AppRoot', () => {
    it('should render AppRoot component', async () => {

        const page = await newSpecPage({
            components: [AppRoot],
            html: `<appRoot></appRoot>`,
        });

        expect(page.root).toEqualHtml(`
      <appRoot></appRoot>
    `);

        expect(page).toBeTruthy();

        expect(page.root).toMatchSnapshot()
    });

});