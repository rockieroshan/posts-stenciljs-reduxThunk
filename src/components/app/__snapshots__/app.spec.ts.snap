// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRoot should render AppRoot component 1`] = `<approot></approot>`;
